### PR TITLE
Pre 1.0 Documentation Work

### PR DESCRIPTION
**Pull request by Cadair**
Still more to do, but got the core of the RTM stuff sorted, need to doc old Events API and webhook setups.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,221 +1,10 @@
 # Matrix-appservice-slack
-A Matrix &lt;--> Slack bridge
+A Matrix &lt;--&gt; Slack bridge
```
Comment (Half-Shot) [inline README.md]: We should have some badges here for things like the matrix room, docker image, etc :)

File: README.md [hunk 2]
```diff
@@ -1,221 +1,10 @@
...
-
-      These arguments can be shortened to single-letter forms:
-
-         ```
-         link -I CHANNELID -R !the-matrix:room.id -t xoxb-xxxxxxxxxx-xxxxxxxxxxxxxxxxxxxx
-         ```
 
+See (the docs)[docs/index.md] for instructions on how to set up the bridge.
 
 ### Legacy - Webhooks
```
Comment (Half-Shot) [inline README.md]: Presumably this, and the other headings will end up under documents?

File: README.md [hunk 3]
```diff
@@ -1,221 +1,10 @@
 # Matrix-appservice-slack
-A Matrix &lt;--> Slack bridge
+A Matrix &lt;--&gt; Slack bridge
```
Comment (Half-Shot) [inline README.md]: We should lay out what features we support here, as a high level description. This probably means specifying that we support webhooks and "rich" bot bridging, as well as puppeting. Might also be nice to have a one liner explaining that you need a homeserver to proceed any further and all that faff.

Comment (Half-Shot) [inline README.md]: A screenshot showing the bridge doing all it's things would be awesome too! [like this](https://github.com/Half-Shot/matrix-appservice-discord#matrix-discord-bridge)

File: docs/datastores.md [hunk 4]
```diff
@@ -41,6 +28,19 @@ db:
 
 Ensure that `dbdir` is not included in the config.
 
+NeDB End-of-life
```
Comment (Half-Shot) [inline docs/datastores.md]: Why shift this down ooi? Shouldn't it be prominent for people who want to upgrade.

File: docs/datastores.md [hunk 5]
```diff
@@ -41,6 +28,19 @@ db:
...
+
+NeDB is a library which is used to store json documents locally to disk to give the bridge some local persistent state. 
+All deployments of this bridge before `1.0` will have been using NeDB.
+
+Starting with version `1.0`, NeDB will be deprecated and shouldn't be used for new installations. NeDB is
+[unmaintained](https://github.com/matrix-org/matrix-appservice-bridge/issues/77) and doesn't scale well for the
+needs of this bridge. Features such as puppeting will not be supported, however existing functionality will continue
+to be maintained until support for NeDB is removed. 
+
+Version 1.0 of the bridge only supports PostgreSQL as an alternative datastore.
```
Comment (Half-Shot) [inline docs/datastores.md]: This contradicts the above statement. I believe we are going to ship 1.0 with support for NeDB as usual, but pushing people to migrate.

File: docs/getting_started.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
+# Getting Started
+
+In this guide we will setup the bridge connected to your homeserver. This guide
+will walk through the most common settings, other options are documented in
+other places.
```
Comment (Half-Shot) [inline docs/getting_started.md]: "other places." needs to be more descriptive.

File: docs/getting_started.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+$ git clone https://github.com/matrix-org/matrix-appservice-slack.git
+$ cd matrix-appservice-slack
+$ npm install
+$ npm run build
+```
+
+## How it Works:
+
+The bridge listens to events using the Slack RTM API over websockets, and to
+matrix events on a port that the homeserver sends events to. This tutorial will
```
Comment (Half-Shot) [inline docs/getting_started.md]: s/matrix/Matrix/

File: docs/getting_started.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+
+## Setup
+
+1. Create a new Matrix room to act as the administration control room. Note its
+   internal room ID (EX: !abcdefg12345hijk:coolserver.com).
+
+2. Decide on a spare local TCP port number to use. It will listen for messages
+   from Matrix and needs to be visible to the homeserver. Take care to configure
+   firewalls appropriately. This ports will be notated as `$MATRIX_PORT` and in
```
Comment (Half-Shot) [inline docs/getting_started.md]: (if your bridge and homeserver aren't colocated)

Comment (Half-Shot) [inline docs/getting_started.md]: -and

File: docs/getting_started.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+   the remaining instructions.
+
+3. Create a `config.yaml` file for global configuration. There is a sample
+   one to begin with in `config/config.sample.yaml` you may wish to copy and
```
Comment (Half-Shot) [inline docs/getting_started.md]:
```suggestion
   one to begin with in `config/config.sample.yaml`. You should copy and
```

File: docs/getting_started.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+   machine you can use `localhost` as the `$HOST` name):
+   
+   ```
+   $ npm start -- -r -c config.yaml -u "http://$HOST:$MATRIX_PORT"
+   ```
+
+6. Start the actual application service.
+
+   ```
+   $ npm start -c config.yaml -p $MATRIX_PORT
```
Comment (Half-Shot) [inline docs/getting_started.md]:
```suggestion
   $ npm start -- -c config.yaml -p $MATRIX_PORT
```

File: docs/getting_started.md [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+8. Invite the bridge bot user into the admin room, so it can actually see and
+   respond to commands. The bot's user ID is formed from the `sender_localpart`
+   field of the registration file, and the homeserver's domain name. For example:
+
+   ```
+   /invite @slackbot:my.server.here
+   ```
+
+NOTE: At the time of writing, Riot does not recognize the Slack bot. This is
+okay. The bot *is there*... probably. Either way, when Riot asks if you're
```
Comment (Half-Shot) [inline docs/getting_started.md]:
```suggestion
okay. The bot *is there*. If Riot asks if you're
```

I'm a grump, but probably is going to confuse folks :p

File: docs/getting_started.md [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+homeserver url without conflicting with each other.
+
+NOTE: If your bridge and homeserver run on different machines, you will need
+to introduce proxying into the mix, which is beyond the scope of this readme.
+There are some really awesome and kind people in the Matrix community. If you're
+ever stuck, you can post a question in the [Matrix Bridging channel]
+(https://matrix.to/#/#bridges:matrix.org).
+
+
+## Setup
```
Comment (Half-Shot) [inline docs/getting_started.md]: We should mention here that we are using Synapse 1.(4?) and Riot 1.(4?)

File: docs/datastores.md (hunk 4 above)
Comment (Cadair) [reply docs/datastores.md]: Everyone, new or upgrading will need the postgres section, and we link to it during the setup guide, so I thought that should be the most prominent. It's probably worth having a specific page on upgrading from pre-1.0

**Comment by Half-Shot** [comment]
FYI, I added my own changes to https://github.com/matrix-org/matrix-appservice-slack/tree/hs/1.0-docs

**Pull request closed without merging**